kythe/go/util/riegeli/writer.go

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2018 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package riegeli
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+)
+
+func (w *Writer) ensureFileHeader() error {
+	if w.fileHeaderWritten {
+		return nil
+	}
+
+	// https://github.com/google/riegeli/blob/master/doc/riegeli_records_file_format.md#file-signature
+	_, err := (&chunk{Header: chunkHeader{ChunkType: fileSignatureChunkType}}).WriteTo(w.w, w.w.pos)
```

Comment:
I think this'd be a little easier to write if you separate the value. Also, given this is static, maybe we can just keep one around?

```
var fileHeader = &chunk{
   Header: chunkHeader{ChunkType: fileSignatureChunkType},
}
…
_, err := fileHeader.WriteTo(w.w, w.w.pos)
```

Reply:
✓